src/models/models.ts

```diff
@@ -0,0 +1,64 @@
+export interface TimerState {
```

Comment:
Think of interfaces as `Types` (because they are _basically_ interchangeable with a Typescript concept called a `Type`, mostly preference whether you use one or another.

In an interface you can define what fields an object will have, and what types are involved with those fields. The basic types are `undefined, null, string, number, void, boolean, any, and probably some others im forgetting`. You use these basic types combined with _structure_ of an interface (you ever seen C code with `struct` ? Same idea) to build up all the types you expect to encounter in your app. 

Try to avoid using `any` since its kind of just a cop out of saying `I have no idea what properties this object will have, so you can assume anything`, which defeats the entire purpose of typescript

Reply:
What's the rationale behind using TypeScript vs. JavaScript when working in React? My guess is that by strongly-typing your variables, you run into fewer bugs?

Reply:
Yup you pretty much nailed it. Not sure what IDE your using (hopefully VSCode or something like that), but with typescript you get the ability to take an object `myObject` and then when you do `myObject.`, your IDE can then suggest and inform you of what the fields are and what you can do. Which as I;m sure you remember in Aaronson's class, switching from BlueJ to Eclipse is like night and day. Same idea with javascript -> typescript

Also actually another MAJOR benefit is just reducing the scope of what you can do, so you cant do ridiculous shit that makes the codebase unmaintainable. Like until you see and work with a truly awful Javascript codebase it can be hard to see just how wrong things can go, but without strong types on your objects its the wild west and it makes you need to write such like "defensive" code (in the sense of well on my object X is _usually_ this property y, but I have no guarantee so if I wanted to do like `x.y.someFunctionOnY()`, I'd have to always check `x.y !== undefined`. With typescript you can use this syntax of `x.y?.someFunctionOnY()` which basically says only call that function if y is defined on X. One of many many useful shortcuts you get with typescript). Basically forces devs to behave and really think about the structure of things